Amity.py

```diff
@@ -0,0 +1,284 @@
+Usage:
+    amity allocaterooms
+    amity viewallocations  [(-r <name_of_room>)]
+    amity viewunallocated
+    amity unallocate (-p <fname> <lname> -r <name_of_room>)
+    amity reset
+    amity (-s | --start)
+    amity (-h | --help | --version)
+Options:
+    -s, --start  Starts the program.
+    -h, --help  Shows a list of commands and their usage.
+"""
+
+from Tkinter import Tk
+import tkFileDialog
+from tkFileDialog import askopenfilename
+import sys
+import cmd
+from docopt import docopt, DocoptExit
+from colorama import init, Fore, Back, Style
+from termcolor import cprint
+from pyfiglet import figlet_format
+from utils.Fileparser import Fileparser
+from db.DatabaseManager import DatabaseManager
+from random import randint
+from utils.Allocations import Allocations
+from models.Rooms import Rooms
+from clint.textui import colored, puts, indent
+from colorama import init
+from termcolor import cprint
+from pyfiglet import figlet_format
+from colorama import init, Fore, Back, Style
+import fileinput
+
+
+# compares the arguments to determine if all have been entered in correct
+# manner
+
+
+def parser(func):
+
+    def fn(self, arg):
+        try:
+            # tries to compare entered commands against the doc
+            opt = docopt(fn.__doc__, arg)
+
+        except DocoptExit as e:
+            # The entered arguments don't match
+
+            print('Sorry,you entered an invalid command')
+            print(e)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+
+"""Class overrides method parser so as to validate input.The input arguments
+    are mapped to respective methods
+"""
+
+
+class Amity (cmd.Cmd):
+
+    prompt = '(Amity): '
+
+    @parser
+    def do_allocaterooms(self, arg):
+        """Usage: allocate """
+
+        allocaterooms(arg)
+
+    def quit(self):
+        self.root.destroy
+
+    @parser
+    def do_viewallocations(self, arg):
+        """Usage: viewallocations [(-r <name_of_room>)]"""
+
+        viewallocations(arg)
+
+    @parser
+    def do_unallocate(self, arg):
+        """Usage: unallocate (-p <fname> <lname> -r <name_of_room>)"""
+
+        unallocate(arg)
+
+    @parser
+    def do_viewunallocated(self, arg):
+        """Usage: viewunallocated """
+
+        viewunallocated(arg)
+
+    @parser
+    def do_reset(self, arg):
+        """Usage: reset """
+
+        reset(arg)
+
+    def do_quit(self, arg):
+        """Exit application."""
+
+        print('Bye!')
+        exit()
+
+opt = docopt(__doc__, sys.argv[1:])
+
+
+"""Resets all allocations
+"""
+
+def reset(docopt_args):
+    puts(colored.green("Are you sure you want to reset allocations? (y)(n)"))
+    answer = raw_input(">")
+    if answer == "y" or answer == "Y":
```

Comment:
**Code Quality: Nitpick**

This condition should have been `if answer.lower() == 'y'`